### ISSUE TITLE
How long does it take to train until getting an FID 1.34

### ISSUE THREAD

**Issue by trungpx** [tzco/Diffusion-wo-CFG#3]
Great work! I really enjoyed reading it.  

Dear authors,
I am trying to run the training on ImageNet. 
Could you share an estimate of the **training time** (in days), the number of **epochs or training steps**, and the **number of GPU**s used? Thanks.

**Comment by mohammadrezanaderi4** [comment]
Dear Authors,
I recently read your paper and had the same question as trungpx. Would you be able to share this information at your earliest convenience?
Looking forward to your response.
Best regards

**Comment by tzco** [comment]
We follow DiT and SiT to train for 1400 epochs (7M iterations) from scratch. When using REPA as initialization, the training can be greatly accelerated and 500K iterations should be enough. In our XL experiments, we mainly use an elastic cluster of A100 GPUs and sometimes change to H100 GPUs if available.

**Comment by mohammadrezanaderi4** [comment]
Thank you!

I was wondering if you used 16 A100 GPUs for training the model. If so, could you estimate the total training time (**in days or hours**) **with and without REPA**, particularly for DiT_XL/2 and SIT_XL/2 **with MG**?

Additionally, as mentioned in the paper, with MG, there is no need to backpropagate the gradients from the unconditional model training. However, you noted that without defining an empty class during training, the model's performance significantly degrades. **Considering the MG term in the loss function and the empty class, would you be able to estimate how much faster training of this model is compared to the CFG method**?

**Comment by tzco** [comment]
I just tested the training code with a local machine with 4 A100 GPUs. The training speed is 0.63 steps per second, so the total training time is about 128 days from scratch and 9 days with initialization.
Also, the training speed is 0.63 with MG loss and 0.84 without it, so the overhead is approximately 1.3x of vanilla CFG training.

**Comment by mohammadrezanaderi4** [comment]
It seems like there might be a slight misunderstanding. The last line in your message says that "the overhead is approximately 1.3x of vanilla CFG training," **but I think you meant to say that your method is 1.3x of vanilla SIT training (without CFG)**.

If we assume that CFG takes about twice the training time of SIT without CFG (i.e., CFG speed is 0.5x vanilla SIT speed), then your method is **35% faster than CFG**, since you’re achieving 1.3x the time for MG compared to the vanilla SIT.

**Does this clarification make sense, or did I miss something in your comparison?**

**Comment by trungpx** [comment]
I think 1.3x slower than baseline SiT in training is reasonable because the MG needs one more forward (totally two) compare to SiT only one forward in training time.

Note that training time with and without cfg of SiT are just the same because cfg only randomly drop class to the dummy class (1000). The total fwd is same.

The advantage of MG is that in the inference it needs only one fwd compared SiT need two. But the price is that the training time of MG required more (1.3x slower as author suggested) compared to SiT training.

> It seems like there might be a slight misunderstanding. The last line in your message says that "the overhead is approximately 1.3x of vanilla CFG training," **but I think you meant to say that your method is 1.3x of vanilla SIT training (without CFG)**.
> 
> If we assume that CFG takes about twice the training time of SIT without CFG (i.e., CFG speed is 0.5x vanilla SIT speed), then your method is **35% faster than CFG**, since you’re achieving 1.3x the time for MG compared to the vanilla SIT.
> 
> **Does this clarification make sense, or did I miss something in your comparison?**

**Comment by mohammadrezanaderi4** [comment]
Thanks @trungpx for the clarification. I was initially thinking that CFG drops classes with a 50% probability, which would almost double the training time to converge. However, after taking a closer look at the paper, I see that they drop classes with a probability of 10-20%. Therefore, as you mentioned, it doesn't add a significant overhead to the model's training.

**Comment by trungpx** [comment]
> Thanks [@trungpx](https://github.com/trungpx) for the clarification. I was initially thinking that CFG drops classes with a 50% probability, which would almost double the training time to converge. However, after taking a closer look at the paper, I see that they drop classes with a probability of 10-20%. Therefore, as you mentioned, it doesn't add a significant overhead to the model's training.

Hello, even with a 10-20% class drop probability, the implementation does not truly "drop" labels but instead reassigns 10% of them (0–999) to a dummy class (1000). This introduces random label noise rather than removing labels. As a result, the training computations for SiT remain identical with and without classifier-free guidance (CFG), which you can verify in the code.

**Comment by mohammadrezanaderi4** [comment]
> > Thanks [@trungpx](https://github.com/trungpx) for the clarification. I was initially thinking that CFG drops classes with a 50% probability, which would almost double the training time to converge. However, after taking a closer look at the paper, I see that they drop classes with a probability of 10-20%. Therefore, as you mentioned, it doesn't add a significant overhead to the model's training.
> 
> Hello, even with a 10-20% class drop probability, the implementation does not truly "drop" labels but instead reassigns 10% of them (0–999) to a dummy class (1000). This introduces random label noise rather than removing labels. As a result, the training computations for SiT remain identical with and without classifier-free guidance (CFG), which you can verify in the code.

Hello @trungpx, and thanks for your follow-up and explanation. As I understand it, CFG combines conditional and unconditional samples during inference to generate a partially conditioned and partially unconditioned image. In MG, however, this process is integrated into the model training by shifting the target during training.

Now, my question is: while both methods define an empty class (or, as you mentioned, a dummy class) to learn the unconditional distribution—and given the critical role of this unconditional data in both methods—do you think that dropping class labels with a probability of 10% to the dummy class can guarantee the model's convergence in the unconditional mode?

Also, do either of these methods calculate the FID of the dummy class separately?

**Comment by trungpx** [comment]
Hello @mohammadrezanaderi4 

> do you think that dropping class labels with a probability of 10% to the dummy class can guarantee the model's convergence in the unconditional mode?

I think yes, empirically some works have demonstrated.

> Also, do either of these methods calculate the FID of the dummy class separately?

No summy class at all. Only FID on real class.

This is a source I found useful when studying CFG some years ago, I am sharing it with you: https://sander.ai/2022/05/26/guidance.html

**Comment by mohammadrezanaderi4** [comment]
Dear @trungpx, this was a highly useful conversation and thank you for sharing the source. If you don’t mind, could we have a short discussion about the different possibilities for advancing these methods?"
